manifests/00-crd-operator-config.yaml

```diff
@@ -13,6 +13,25 @@ spec:
   subresources:
     status: {}
   versions:
-  - name: v1
-    served: true
-    storage: true
+    - name: v1
+      served: true
+      storage: true
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            authentication:
+              properties:
+                logoutRedirect:
+                  pattern: ^((http|https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))$
+                  type: string
+            customization:
+              properties:
+                documentationBaseURL:
+                  #pattern: ^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+\/$
+                  pattern: ^((http|https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))\/$
```

Comment:
@enj has had several objections to the trailing `/` for `documentationBaseURL`.  
@spadgett fyi 
Not sure if we want to chat about that now before we add validation for it.

Reply:
The documentation URL currently is required to end with `/` since we append topics to it. It will prevent users from adding a URL like `https://docs.okd.io/latest/welcome/index.html` which would break our links.

I think we need to keep the trailing slash.

Reply:
For the top level config `consoleURL` I will require the following validation (because the rules are simple and precise to explain - and they make it easy to consume the string):

1. no error from `url.Parse`
1. scheme must be `https`
1. domain must be valid
1. no path
1. no query
1. no fragment

This means no trailing slash (as that means path = `/`).  It makes sense to make the other URL validation consistent (up to you to decide if it really makes sense to allow `http` scheme on the other URLs) across all configs.  Thus when you append, always do `documentationBaseURL` + `/` + `topic` (I think that is what you are saying you need to be able to do).

Reply:
But then we don't catch the case where the user puts in `/index.html` which becomes `/index.html/<topic>`. The point of requiring the trailing `/` is to make it clear to the admin changing this value that it must be a directory and failing validation if they don't, instead of having broken URLs in the UI.

Reply:
So I think we can resolve this.

Reply:
Based on talking on slack with @spadgett and @benjaminapetersen I see the requirements for `documentationBaseURL` to be:

1. no error from `url.Parse`
1. scheme must be `https`
1. domain must be valid
1. **path must be non-empty and must end in slash**
1. no query
1. no fragment

Reply:
(I realize that you have to use regex hacks and cannot use `url.Parse`)

Reply:
Looks right, thanks

Who are we describing to? If it's the person editing the config, "no error from `url.Parse`" probably isn't meaningful

Reply:
Right, unfortunately when this fails all it will do is dump the regex, which will not clearly describe any of that. 😢

Reply:
At least we could put something in the doc so that `oc explain` or the help topic describes what it needs to be. Agree the regex error message is not a good user experience.

Reply:
@enj - As for forcing https, I am worried about testing when you have no cert.  For doc link, it might be fine, but logout could easily not have an https option.

Reply:
The logout url presumably redirects you to a page where you can manage your SSO session.  How would it ever be safe if it was http?

Minting a CA/certs in tests, while annoying, is not hard.

Reply:
I certainly agree that I would not want people to use unsecure methodolgies, but I was more just stating there are occasions were having the option of http would be nice.  I run the operator/console locally on libvirt and I would prefer to not have to setup a cert.  That said, I am just trying to debate the state of the internet and how people operate on whether we can safely assume that no one will use http.  While I would not recommend it, could a customer want to setup an openshift cluster on-prem inside their private network that has no certs for their developers?  I would hate to make things so restrictive as to discourage people from using or trying okd.

Reply:
@spadgett @benjaminapetersen - Do you think I should limit this to only https for now?  I am fine either way.

Reply:
My observation is that we tend to say we are secure by default, so I'd lean toward keeping the `s`.

Reply:
+1 to https. We're not able to make it more restrictive later, so I'd prefer https for now.

Reply:
Unsecure http has been removed